### PR TITLE
Support Force-update in Mirror and improve Tracing in mirror

### PR DESCRIPTION
**Pull request by zeripath**
This PR allows the mirror service to handle force-updates and improves Trace logging within the mirror service considerably.

It also fixes a bug in `log.NewColoredIDValue()` which led to a double indirection and incorrect IDs being printed out.

### CODE REVIEW

File: services/mirror/mirror.go [hunk 1]
```diff
@@ -157,6 +157,26 @@ func parseRemoteUpdateOutput(output string) []*mirrorSyncResult {
 				refName:     refName,
 				newCommitID: gitShortEmptySha,
 			})
+		case strings.HasPrefix(lines[i], " + "): // Force update
+			idx := strings.Index(refName, " ")
+			if idx > -1 {
+				refName = refName[:idx]
+			}
```
Comment (lafriks) [inline services/mirror/mirror.go]:
```suggestion
			if idx := strings.Index(refName, " "); idx > -1 {
				refName = refName[:idx]
			}
```

**Comment by lafriks** [comment]
small nit but otherwise lgtm

File: modules/log/colors.go [hunk 2]
```diff
@@ -355,7 +355,7 @@ func NewColoredValueBytes(value interface{}, colorBytes *[]byte) *ColoredValue {
 // The Value will be colored with FgCyan
 // If a ColoredValue is provided it is not changed
 func NewColoredIDValue(value interface{}) *ColoredValue {
-	return NewColoredValueBytes(&value, &fgCyanBytes)
+	return NewColoredValueBytes(value, &fgCyanBytes)
```
Comment (lunny) [inline modules/log/colors.go]: Maybe a seperate PR so that it could be backport easily.

Comment (zeripath) [reply modules/log/colors.go]: See #12294

**Comment by 6543** [comment]
@zeripath there should be an option to allow force-update on mirrors since it could be an eavel atack to delete code

some mirrors exist to have a copy of some code (prevent code erase of eavil actions witch could happen on opencource repos)

**Comment by zeripath** [comment]
@6543 git is doing the mirroring not us - so the force update already happens - the issue is that the gitea UI doesn't get updated.

**Comment by 6543** [comment]
ok konssider this pull as bugfix and my comment as new feature-request

**Review by mrsdizzie — APPROVED** [review]

**Review by techknowlogick — APPROVED** [review]

**Pull request merged**